docs/development-environment.md

```diff
@@ -1,298 +1,444 @@
 
-```
-svn co https://develop.svn.wordpress.org/trunk/ /tmp/wordpress-develop
-cd /tmp/wordpress-develop
-cp wp-tests-config-sample.php wp-tests-config.php
-```
+<a name="jetpack-development"></a>
+# Development workflow
 
-Set the database information for your testing DB in the file `/tmp/wordpress-develop/wp-tests-config.php`. You may need to create this database.
+To start work on the Jetpack plugin you need to follow these steps:
 
-To run tests on your machine, you can run `phpunit` while in the Jetpack directory.
+1. [Clone the repository](#jetpack-development-repository)
+2. [Install the development tools](#setup-development-tools)
+3. Make sure Jetpack is enabled on your WordPress site
+4. [Build Jetpack](#jetpack-development-build)
+5. Open `/wp-admin/admin.php?page=jetpack` in your browser.
 
-To run WooCommerce integration tests, you'll need the WooCommerce plugin installed alongside Jetpack (in `../woocommerce`), and you can run:
+<a name="jetpack-development-repository"></a>
+## Clone the repository
 
+Make sure you have `git`, `node`, `yarn`, and a working WordPress installation.
+Clone this repository inside your Plugins directory.
+	
+```sh
+git clone git@github.com:Automattic/jetpack.git
+cd jetpack
 ```
-JETPACK_TEST_WOOCOMMERCE=1 phpunit
-```
-
-To run multisite tests, run:
+	
+ You'll need to have a public SSH key setup with GitHub, which is more secure than saving your password in your keychain.
+ There are more details about [setting up a public key on GitHub.com](https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account).
 
-```
-phpunit -c tests/php.multisite.xml
-```
+<a name="jetpack-development-build"></a>
+## Building Jetpack
 
-To filter and run just a particular test, you can run:
+To work on Jetpack you need to build the JavaScript and CSS components of the plugin's admin interface. This will generate the run time bundle (`_inc/build/admin.js`)
 
-```
-phpunit --filter my_test_name
-```
+There are three types of builds:
 
-If you're not familiar with PHP Unit Testing, you can also check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/)
+* ### Development build
+	The standard development build will create unminified versions of the JavaScript and CSS files. To build Jetpack like this run:
+	
+	```sh
+	yarn build
+	```
+	
+* ### Continuous Development build
+	By default the development build above will run once and if you change any of the files, you need to run `yarn build` again to see the changes on the site. If you want to avoid that, you can run a continous build that will rebuild anytime it sees any changes on your local filesystem. To run it, use:
+	
+	```sh
+	yarn watch
+	```	
+	
+* ### Production build
+	The production build will generate minified files without duplicated code (resulting from dependencies) and will also generate the matching source map and language files. To build it use:
+	
+	```sh
+	NODE_ENV=production yarn build-client
```

Comment:
We can shorten that to `yarn build-production-client`.

Reply:
Thanks! Updated.